shielded-pool/src/component.rs

```diff
@@ -246,6 +248,29 @@ impl Component for ShieldedPool {
             .unwrap();
         }
 
+        // Close the block in the TCT
+        self.tiered_commitment_tree
+            .end_block()
+            .expect("ending TCT block should never fail");
```

Comment:
If we `expect` here, we'll halt `pd` in case of an invariant violation. We'll want that eventually, but initially, we want the TCT to be a sidecar that gets exercised but not fed into consensus.

Instead, I we should match on the result and do a `tracing::error!` on failure, but otherwise continue executing.